Source/Utils/WarpXUtil.cpp

```diff
@@ -446,6 +469,21 @@ void ReadBCParams ()
             }
 
         }
+        if (WarpX::field_boundary_lo[idim] == FieldBoundaryType::Damped ||
+            WarpX::field_boundary_hi[idim] == FieldBoundaryType::Damped ) {
+#ifdef WARPX_USE_PSATD
```

Comment:
Do we need the `#ifdef WARPX_USE_PSATD` switch in this case (given that there is nothing that cannot compile without PSATD)?

Could we maybe simply use the runtime switch `if (maxwell_solver_id == MaxwellSolverAlgo::PSATD)` and do something like the following?

```cpp
if (WarpX::field_boundary_lo[idim] == FieldBoundaryType::Damped ||
    WarpX::field_boundary_hi[idim] == FieldBoundaryType::Damped )
{
    int maxwell_solver_id = GetAlgorithmInteger(pp_algo, "maxwell_solver");
    if (maxwell_solver_id == MaxwellSolverAlgo::PSATD)
    {
        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
            WarpX::field_boundary_lo[idim] == WarpX::field_boundary_hi[idim],
            "field boundary in both lo and high must be set to Damped for PSATD");
    }
    else // FDTD
    {
        amrex::Abort("FieldBoundaryType::Damped is not supported for FDTD");
    }
}
```

Do you think this could work?

Reply:
Yes this could work. I actually wanted to ensure that if someone has set maxwell_solver_id to PSATD but not compiled with PSATD, this would catch it. Although there are other places in the code where this could be caught as well.

Reply:
I think that this kind of check would be better done in `WarpX::ReadParameters` where the input specifying which field solver is being used is already being read in. This avoids duplication of code and puts it in the same place where other similar checks on input are being done. My understanding is that this routine is only needed here to ensure that the input parameter is_periodic is setup properly when it is passed to amr.